lib/spack/spack/patch.py

```diff
@@ -231,16 +319,34 @@ def stage(self):
         )
         return self._stage
 
-    def to_dict(self):
+    def to_dict(self) -> Dict[str, Any]:
+        """Dictionary representation of the patch.
+
+        Returns:
+            A dictionary representation.
+        """
```

Comment:
I think we can avoid, or shorten, trivial docstrings :slightly_smiling_face: There are a couple more in this PR.

Reply:
I personally disagree, but that's partly because I religiously use pydocstyle, which requires me to always add those. If you don't feel strongly, I would like to keep and encourage these.

Reply:
I’m with @adamjstewart  on this one

Reply:
I'm more in line with Robert C. Martin "Clean code" suggestion to avoid redundant or mandated comment. But I can live with collapsing them. 

The book example for "Noise comments":
```java
/**
 * Return the day of the month
 *
 * @return the day of the month
 */
public int getDayOfTheMonth() {
    return dayOfMonth;
}
```
doesn't look that different from this one :slightly_smiling_face:

Reply:
You're right, my code is the definition of a noise comment. I'm not sure how to reconcile my inner desires to placate pydocstyle and please Robert C. Martin. I'll let @tgamblin decide for me.

Reply:
can't you let your editor generate trivial comments

Reply:
Nope, not that I know of. It's also quite broken, I've been meaning to fix it for years.

Reply:
ok if everyone agrees I'm fine with removing these comments, though not sure we need to do it now. I don't have a strong opinion on it.